metis-enrichment/metis-enrichment-service/src/main/java/eu/europeana/enrichment/utils/MongoDatabaseUtils.java

```diff
@@ -207,22 +211,22 @@ public static boolean dbExists(String host, int port) {
   /**
    * Find TermList by codeURI
    */
-  public static MongoTermList findByCode(String codeUri, String dbtable) {
-    switch (dbtable) {
-      case "concept":
+  public static MongoTermList<?> findByCode(String codeUri, EntityClass entityClass) {
+    switch (entityClass) {
```

Comment:
Sonar complains about the number of returns. I suppose initializing the MongoTermList and then returning once should work.

Reply:
Fixed.